src/PowerPing/Ping.cs

```diff
@@ -410,24 +417,39 @@ private PingResults SendICMP(PingAttributes attrs)
                         if (rnd.Next(20) == 1) { throw new SocketException(); }
                     }
 
-                    // Wait for response
-                    byte[] buffer = new byte[attrs.RecieveBufferSize]; // Ipv4Header.length + IcmpHeader.length + attrs.recievebuffersize
-                    bytesRead = sock.ReceiveFrom(buffer, ref ep);
-                    responseTimer.Stop();
-
-                    // Store reply packet
-                    ICMP response = new ICMP(buffer, bytesRead);
+                    ICMP response;
+                    TimeSpan replyTime;
+                    do {
+                        // Wait for response
+                        bytesRead = sock.ReceiveFrom(receiveBuffer, ref ep);
+                        replyTime = new TimeSpan(Helper.StopwatchToTimeSpanTicks(Stopwatch.GetTimestamp() - requestTimestamp));
```

Comment:
This calculation (inside of StopwatchToTimeSpanTicks) seems to be the most performance intense line of the whole function. If we can reduce the complexity of this function even a bit it would be nice, seeing as this is basically the core of the program.

Reply:
I'm surprised there would be a performance issue there; if the profiler said that I wonder if it's just a misleading result or the JITer didn't inline the method if it was in debug mode. Anyway, I cached one of the calculations just in case.

Reply:
Yeah it wasn't a big performance increase really. I think the inline you have done is nicer, any thing that we can slim down in this sending loop the better.